tests-scan

```diff
@@ -88,7 +89,7 @@ def main():
 
         # When run from c-p/bots checkout without PR number or PR data we want to scan also all
         # external projects. E.g. `./tests-scan` in c-p/b checkout will scan all projects
-        if opts.repo == "cockpit-project/bots" and not opts.pull_data and not opts.pull_number:
+        if not repo_specified and opts.repo == "cockpit-project/bots" and not opts.pull_data and not opts.pull_number:
```

Comment:
This is another hack, though. At first sight it's not clear why  this should differ between an implied default and an explicit `--repo cockpit-project/bots`. I thought this mainly needs to differentiate between a tests-scan that got triggered in response to a "PR created" event (in which case we want external tests) and one in response to a "statuses" event (in which case we don't). So please add a `# HACK` at least, so that we can clean this up later on, and add a detailled commit message.

Reply:
It indeed is - but it is due to another hack being there - the one where we set up `opts.repo`. If we would not do that, we could just check if `opts.repo` is defined and then check `api.repo`. I'll add there comment explaining it.

Reply:
Done

Reply:
That's not really what I meant. Of course the introduction of `repo_specified` is a shallow hack around changing `opts.repo`. But my concern is conceptual, not syntactical. I meant that running external tests or not should not depend on whether `--repo` is given on the command line or not, but for which GitHub event it is being called.

Reply:
So as talked on IRC - this was misunderstanding. And we decided that we don't really need the concept of scanning of external projects and webhook can do it itself. So I changed that commit to drop this part all together.

Reply:
Thanks, this is much better indeed! @Gundersanne, can you please review again?